airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml

```diff
@@ -0,0 +1,404 @@
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "scorecards"
+  sources_stream:
+    type: DeclarativeStream
+    $options:
+      name: "sources"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "sources"
+  users_stream:
+    type: DeclarativeStream
+    $options:
+      name: "users"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "users"
+  demographics_answers_stream:
+    type: DeclarativeStream
+    $options:
+      name: "demographics_answers"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "demographics/answers"
+  demographics_answer_options_stream:
+    type: DeclarativeStream
+    $options:
+      name: "demographics_answer_options"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "demographics/answer_options"
+streams:
+  - "*ref(definitions.applications_stream)"
+  - "*ref(definitions.applications_demographics_answers_stream)"
+  - "*ref(definitions.applications_interviews_stream)"
+  - "*ref(definitions.candidates_stream)"
+  - "*ref(definitions.close_reasons_stream)"
+  - "*ref(definitions.custom_fields_stream)"
+  - "*ref(definitions.degrees_stream)"
+  - "*ref(definitions.demographics_answers_stream)"
+  - "*ref(definitions.demographics_answer_options_stream)"
+  - "*ref(definitions.questions_stream)"
+  - "*ref(definitions.demographics_answers_answer_options_stream)"
+  - "*ref(definitions.demographics_question_sets_stream)"
+  - "*ref(definitions.demographics_question_sets_questions_stream)"
+  - "*ref(definitions.departments_stream)"
+  - "*ref(definitions.jobs_stream)"
+  - "*ref(definitions.jobs_openings_stream)"
+  - "*ref(definitions.interviews_stream)"
+  - "*ref(definitions.job_posts_stream)"
+  - "*ref(definitions.job_stages_stream)"
+  - "*ref(definitions.jobs_stages_stream)"
+  - "*ref(definitions.offers_stream)"
+  - "*ref(definitions.rejection_reasons_stream)"
+  - "*ref(definitions.scorecards_stream)"
+  - "*ref(definitions.sources_stream)"
+  - "*ref(definitions.users_stream)"
+
+check:
+  type: CheckStream
+  stream_names: ["applications"]
```

Comment:
Greenhouse is a permissioned system, and the user may only have access to certain streams based on the scopes available to them: https://support.greenhouse.io/hc/en-us/articles/115000521723-Manage-Harvest-API-key-permissions

This check is no worse than the current one. But it would be interesting to think through what it would take to support this in the CDK. maybe it requires something like https://github.com/airbytehq/airbyte/issues/14908